aws_embedded_metrics/logger/metrics_context.py

```diff
@@ -66,6 +68,10 @@ def put_dimensions(self, dimensions: Dict[str, str]) -> None:
             # TODO add ability to define failure strategy
             return
 
+        if len(dimensions) > MAX_DIMENSION_SET_SIZE:
+            raise DimensionSetExceededError(
```

Comment:
Suggestion: Can we put this code behind a validateDimensionSet method? That way you can avoid duplication of this check and the exception.